app/src/main/java/com/marvelapp/presentation/activity/SplashActivity.kt

```diff
@@ -0,0 +1,52 @@
+package com.marvelapp.presentation.activity
+
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.view.animation.AnimationUtils
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.marvelapp.R
+import com.marvelapp.data.MarvelStates
+import com.marvelapp.databinding.ActivitySplashBinding
+import com.marvelapp.presentation.viewModel.SplashViewModel
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class SplashActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySplashBinding
+    private val splashViewModel: SplashViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySplashBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        splashViewModel.startAnimation()
+        binding.apply {
+            startAnimation()
```

Comment:
this should be called from a updateUI method or kind of, something that listens the LiveData or StateFlow from the viewmodel

Reply:
Solved